sdks/java/core/src/main/java/org/apache/beam/sdk/options/PipelineOptionsFactory.java

```diff
@@ -1398,42 +1428,59 @@ public int compare(Method o1, Method o2) {
   /**
    * A {@link Predicate} that returns true if the method is annotated with {@code annotationClass}.
    */
-  static class AnnotationPredicate implements Predicate<Method> {
-    static final AnnotationPredicate JSON_IGNORE = new AnnotationPredicate(
+  static class AnnotationPredicates {
+    static final AnnotationPredicates JSON_IGNORE = new AnnotationPredicates(
+        JsonIgnore.class,
+        new Predicate<Annotation>() {
+          @Override
+          public boolean apply(@Nonnull Annotation input) {
+            return input.annotationType().equals(JsonIgnore.class);
```

Comment:
use instanceof or swap the direction so its JsonIgnore.class.equals(input.annotationType)